src/utils/walletConnect.ts

```diff
@@ -261,3 +321,169 @@ export function onSessionProposal(
     getActiveRoute()?.name === Routes.WALLET_CONNECT_APPROVAL_SHEET
   );
 }
+
+export async function onSessionRequest(
+  event: SignClientTypes.EventArguments['session_request']
+) {
+  const client = await signClient;
+
+  logger.debug(`WC v2: session_request`, { event });
+
+  const { id, topic } = event;
+  const { method, params } = event.params.request;
+
+  if (isSigningMethod(method)) {
+    // transactions aren't a `[address, message]` tuple
+    const isTransactionMethod = isTransactionDisplayType(method);
+    let { address, message } = parseRPCParams({ method, params });
+    const allWallets = store.getState().wallets.wallets;
+
+    if (!isTransactionMethod) {
+      if (!address || !message) {
+        logger.error(
+          new RainbowError(
+            `WC v2: session_request exited, no address or messsage`
+          ),
+          {
+            address,
+            message,
+          }
+        );
+
+        await client.respond({
+          topic,
+          response: formatJsonRpcError(id, `Invalid RPC params`),
+        });
+
+        return;
+      }
+
+      // for TS only, should never happen
+      if (!allWallets) {
+        logger.error(new RainbowError(`WC v2: allWallets is null`));
+        return;
+      }
+
+      const selectedWallet = findWalletWithAccount(allWallets, address);
+
+      if (!selectedWallet || selectedWallet?.type === WalletTypes.readOnly) {
+        logger.debug(
+          `WC v2: session_request exited, selectedWallet was falsy or read only`
+        );
+
+        await client.respond({
+          topic,
+          response: formatJsonRpcError(id, `Wallet is read-only`),
+        });
+
+        return;
+      }
+    } else {
+      address = params[0].from;
```

Comment:
Unsure where to pull the "signing address" from in this case. Will pull someone aside for this one offline.